src/ucp/core/ucp_worker.c

```diff
@@ -3018,8 +3033,6 @@ void ucp_worker_keepalive_remove_ep(ucp_ep_h ep)
         return;
     }
 
-    ucs_assert(!ucs_list_is_empty(&worker->all_eps));
```

Comment:
why removed?
if ep is on the list IMO it should be non-empty?

Reply:
new test produces the case when `ucp_worker_keepalive_remove_ep` can be called twice for the same EP - from ep_close(force) and later, when err handling is processed and EP discarded (ref count == 0), as discussed with @hoopoepg this function only checks if the iterator does not point to  the ep and advances if it does. so the assert is redundant

Reply:
yes, I agree that EP should be on the list when we removing it from KA

Reply:
I guess currently we don't have a way to know if the ep on the list or not

Reply:
add comment for this?